tchannel-core/src/test/java/com/uber/tchannel/tracing/TracingContextTest.java

```diff
@@ -22,81 +22,16 @@
 package com.uber.tchannel.tracing;
 
 import com.uber.jaeger.Tracer;
-import com.uber.jaeger.reporters.InMemoryReporter;
-import com.uber.jaeger.samplers.ConstSampler;
-import com.uber.jaeger.samplers.Sampler;
-import io.opentracing.Span;
-import org.junit.After;
-import org.junit.Before;
-import org.junit.Test;
+import org.jetbrains.annotations.NotNull;
 
-import java.util.EmptyStackException;
-
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
-
-public class TracingContextTest {
-    private Tracer tracer;
-    private InMemoryReporter reporter;
-    private TracingContext tracingContext;
-
-    @Before
-    public void setUp() throws Exception {
-        reporter = new InMemoryReporter();
-        Sampler sampler = new ConstSampler(true);
-        tracer = new Tracer.Builder("tchannel-name", reporter, sampler).build();
-
-        tracingContext = new TracingContext.ThreadLocal();
-    }
-
-    @After
-    public void tearDown() {
-        reporter.close();
-    }
-
-    @Test(expected = EmptyStackException.class)
-    public void testTracingContextNoCurrent() throws Exception {
-        tracingContext.currentSpan();
-    }
-
-    @Test(expected = EmptyStackException.class)
-    public void testTracingContextCannotPop() throws Exception {
-        tracingContext.popSpan();
-    }
-
-    @Test
-    public void testTracingContext() throws Exception {
-        assertFalse(tracingContext.hasSpan());
-
-        Span span = tracer.buildSpan("test").startManual();
-        tracingContext.pushSpan(span);
-        assertTrue(tracingContext.hasSpan());
-        assertEquals(span, tracingContext.currentSpan());
-        assertEquals(span, tracingContext.popSpan());
-        assertFalse(tracingContext.hasSpan());
+/**
+ * Tests {@link TracingContext.ThreadLocal} implementation of {@link TracingContext} interface.
+ */
+public class TracingContextTest extends TracingContextTestBase {
```

Comment:
main tests functionality was extracted into the shared `TracingContextTestBase` parent class.